SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -222,6 +222,23 @@ JUPYTER_PATH=${ENV_PATH}/share/jupyter python -m ipykernel install --name "${ENV
 # We modify the already existing Python3 kernel with the kernel.json of the environment
 ln -f -s ${ENV_PATH}/share/jupyter/kernels/${ENV_NAME}/kernel.json /home/$USER/.local/share/jupyter/kernels/python3/kernel.json | tee -a ${LOG_FILE}
 
+# Pass Spark configurations to the environment kernel so it can be used in the notebooks
+if [ -n "${INSTALL_NXCALS}" ]; then
+    export SPARK_HOME="${ENV_PATH}/nxcals-bundle"
+    export PYSPARK_PYTHON="./environment/bin/python"
+    echo -e "export SPARK_HOME=\"${SPARK_HOME}\"\nexport PYSPARK_PYTHON=\"${PYSPARK_PYTHON}\"" >> /home/$USER/.bash_profile
+
+    # Passing configs to the kernel.json
+    export KERNEL_JSON=${ENV_PATH}/share/jupyter/kernels/${ENV_NAME}/kernel.json
+    jq --arg SPARK_HOME "${SPARK_HOME}" \
+       --arg PYSPARK_PYTHON "${PYSPARK_PYTHON}" \
+       --arg PATH "${PATH}" \
+       --arg VIRTUAL_ENV "${ENV_PATH}" \
+       '. + {env: {$SPARK_HOME, $PYSPARK_PYTHON, $PATH, $VIRTUAL_ENV}}' \
+       ${KERNEL_JSON} > /tmp/temp_kernel.json
```

Comment:
Do you really need a temporary or you can just write directly into KERNEL_JSON?

Reply:
I tried and the file becomes empty. I believe it may be reading/writing the file at the same time

Reply:
Ok, then create the temporary file with `mktemp`.